File: srcha/Entity/Produit.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\QueryBuilder;

/**
 * Produit
 *
 * @ORM\Table(name="produit", indexes={@ORM\Index(name="idcategorie", columns={"idcategorie"})})
 * @ORM\Entity
 */
class Produit
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="nomproduit", type="string", length=255, nullable=false)
     */
    private $nomproduit;

    /**
     * @var float
     *Assert/NotEqualTo(
     * value=0,
     * message="le prix d'un produit ne doit pas etre égal à 0"
     * )
     * @ORM\Column(name="prixproduit", type="float", precision=10, scale=0, nullable=false)
     */
    private $prixproduit;

    /**
     * @var int
     * * @Assert\Positive
     * @ORM\Column(name="stockproduit", type="integer", nullable=false)
     */
    private $stockproduit;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="string", length=255, nullable=false)
     */
    private $description;

    /**
     * @var string
     *
     * @ORM\Column(name="photo", type="string", length=255, nullable=false)
     * @Assert\NotBlank(message="Please, upload the file")
     * @Assert\File(mimeTypes={ "image/jpeg" })

     */
    private $photo;

    /**
     * @var \Categorie
     *
     * @ORM\ManyToOne(targetEntity="Categorie")
     * @ORM\JoinColumns({
     * @ORM\JoinColumn(name="idcategorie", referencedColumnName="idc")
     * })
     */
    private $idcategorie;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomproduit(): ?string
    {
        return $this->nomproduit;
    }

    public function setNomproduit(string $nomproduit): self
    {
        $this->nomproduit = $nomproduit;

        return $this;
    }

    public function getPrixproduit(): ?float
    {
        return $this->prixproduit;
    }

    public function setPrixproduit(float $prixproduit): self
    {
        $this->prixproduit = $prixproduit;

        return $this;
    }

    public function getStockproduit(): ?int
    {
        return $this->stockproduit;
    }

    public function setStockproduit(int $stockproduit): self
    {
        $this->stockproduit = $stockproduit;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getPhoto()
    {
        return $this->photo;
    }

    public function setPhoto($photo): self
    {
        $this->photo = $photo;

        return $this;
    }

    public function getIdcategorie(): ?Categorie
    {
        return $this->idcategorie;
    }

    public function setIdcategorie(?Categorie $idcategorie): self
    {
        $this->idcategorie = $idcategorie;

        return $this;
    }




    /**
     * @inheritDoc
     */
    public function jsonSerialize()
    {
        return get_object_vars($this);
    }
}

File: myapp/entities/Produit.java

package com.mycompany.myapp.entities;

import com.codename1.ui.TextField;
import java.util.Date;


public class Produit {
   
    private int id;

    private String nomproduit;
    private Float prixproduit;
    private int idcategorie=49;
    private String description ;
    private int stockproduit ;
    private String photo;

    public Produit() {
        
    }

  

    public Produit(String nomproduit, Float prixproduit, Integer stockproduit, String description) {
         this.nomproduit = nomproduit;
        this.prixproduit = prixproduit;
      
        this.description = description;
        this.stockproduit = stockproduit;
    }

 

    public int getId() {
        return id;
    }

    public Produit(String nomproduit, Float prixproduit, int idcategorie, String description, int stockproduit, String photo) {
        this.nomproduit = nomproduit;
        this.prixproduit = prixproduit;
        this.idcategorie = idcategorie;
        this.description = description;
        this.stockproduit = stockproduit;
        this.photo = photo;
    }

    public Produit(int id, String nomproduit, Float prixproduit, int idcategorie, String description, int stockproduit, String photo) {
        this.id = id;
        this.nomproduit = nomproduit;
        this.prixproduit = prixproduit;
        this.idcategorie = idcategorie;
        this.description = description;
        this.stockproduit = stockproduit;
        this.photo = photo;
    }
    

    public Produit(String nomproduit, Float prixproduit, String description, int stockproduit, String photo) {
        this.nomproduit = nomproduit;
        this.prixproduit = prixproduit;
        this.description = description;
        this.stockproduit = stockproduit;
        this.photo = photo;
    }
    
    
    

    public void setId(int id) {
        this.id = id;
    }

    public String getNomproduit() {
        return nomproduit;
    }

    public void setNomproduit(String nomproduit) {
        this.nomproduit = nomproduit;
    }

    public Float getPrixproduit() {
        return prixproduit;
    }

    public void setPrixproduit(Float prixproduit) {
        this.prixproduit = prixproduit;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getStockproduit() {
        return stockproduit;
    }

    public void setStockproduit(int stockproduit) {
        this.stockproduit = stockproduit;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }

    public int getIdcategorie() {
        return idcategorie;
    }

    public void setIdcategorie(int idcategorie) {
        this.idcategorie = idcategorie;
    }

    @Override
    public String toString() {
        return "Produit{" + "id=" + id + ", nomproduit=" + nomproduit + ", prixproduit=" + prixproduit + ", idcategorie=" + idcategorie + ", description=" + description + ", stockproduit=" + stockproduit + ", photo=" + photo + '}';
    }
    

   

    



    
}
